src/GovUk.Education.ExploreEducationStatistics.Admin/Security/AuthorizationHandlers/UpdateSpecificMethodologyAuthorizationHandlers.cs

```diff
@@ -1,18 +1,56 @@
+using System.Linq;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Admin.Services.Interfaces.Methodologies;
+using GovUk.Education.ExploreEducationStatistics.Common.Services.Security;
+using GovUk.Education.ExploreEducationStatistics.Content.Model;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.Database;
 using Microsoft.AspNetCore.Authorization;
-using static GovUk.Education.ExploreEducationStatistics.Admin.Security.SecurityClaimTypes;
+using Microsoft.EntityFrameworkCore;
 
 namespace GovUk.Education.ExploreEducationStatistics.Admin.Security.AuthorizationHandlers
 {
     public class UpdateSpecificMethodologyRequirement : IAuthorizationRequirement
     {
     }
 
-    public class
-        UpdateSpecificMethodologyAuthorizationHandler : HasClaimAuthorizationHandler<
-            UpdateSpecificMethodologyRequirement>
+    public class UpdateSpecificMethodologyAuthorizationHandler
+        : CompoundAuthorizationHandler<UpdateSpecificMethodologyRequirement, Methodology>
     {
-        public UpdateSpecificMethodologyAuthorizationHandler() : base(UpdateAllMethodologies)
+        public UpdateSpecificMethodologyAuthorizationHandler(IMethodologyRepository methodologyRepository)
+            : base(
+                new UpdateAllSpecificMethodologiesAuthorizationHandler(),
+                new HasNonPrereleaseRoleOnAnyAssociatedReleaseAuthorizationHandler(methodologyRepository)) {}
+    }
+
+    public class UpdateAllSpecificMethodologiesAuthorizationHandler
```

Comment:
Man authorization handler names are sooooo verbose!

Would consider making these dependent authorization handlers nested in `UpdateSpecificMethodologyAuthorizationHandler`. This way you can drop the `AuthorizationHandler` part of the class name, as it's already implied from the context. I've done a similar thing in `LegacyReleaseAuthorizationHandlers`.